### PR TITLE
Fix multiple bugs with user status

### PR DESCRIPTION
**Pull request by CarlSchwan**
* Fix editing the status while on a call so that we don't send a bogus request to the server
* Clean backup user status when setting a new status. This fixes one issue there the backup status was not correctly removed and when the user was trying to set a new one, it aborded because there was already one in the database.
* Don't try to revert then there is nothing to revert
* A bit more type hinting

Signed-off-by: Carl Schwan <carl@carlschwan.eu>

### CODE REVIEW

File: apps/user_status/lib/Controller/UserStatusController.php [hunk 1]
```diff
@@ -99,6 +99,7 @@ public function getStatus(): DataResponse {
 	public function setStatus(string $statusType): DataResponse {
 		try {
 			$status = $this->service->setStatus($this->userId, $statusType, null, true);
+			$this->service->removeUserStatus($this->userId, true);
```
Comment (PVince81) [inline apps/user_status/lib/Controller/UserStatusController.php:102]: so we set the status and remove it right away ?
from someone not familiar with the internals it looks a bit confusing, wonder if we should add PHP comments here to clarify this apparent contradiction ?

Comment (CarlSchwan) [reply apps/user_status/lib/Controller/UserStatusController.php]: Done, the `true` was indicating that the status to remove was a backup status. I changed that by creating a new method: removeBackupStatus, this is more clear.

File: apps/user_status/lib/Service/StatusService.php [hunk 2]
```diff
@@ -346,9 +346,21 @@ public function clearMessage(string $userId): bool {
...
+		} catch (DoesNotExistException $ex) {
+			// if there is no status to remove, just return
+			return false;
+		}
+
+		$this->mapper->delete($userStatus);
+		return true;
+	}
+
+	public function removeBackupUserStatus(string $userId): bool {
```
Comment (PVince81) [inline apps/user_status/lib/Service/StatusService.php:361]: phpdoc maybe ? :wink:

**Review by PVince81 — APPROVED** [review]
:+1: thanks for the update, it's much clearer now

File: apps/user_status/lib/Service/StatusService.php (hunk 2 above)
Comment (CarlSchwan) [reply apps/user_status/lib/Service/StatusService.php:361]: The argument and return value are already type-hinted and the method name is hopefully clear enough that this doesn't need more explanation.

**Review by nickvergessen — APPROVED** [review]
Seems to work better now

**Pull request merged**